### PR TITLE
Fix DMI Neumann boundary conditions for non-3D simulations

### PR DESCRIPTION
**Pull request by ilateur**
This should fix issue #352. Calculation of the effective magnetic field of bulk DMI was skipped when there was no derivative to be taken in the z-direction, but the Neumann boundary conditions do not cancel out. Open BC can still be skipped, and for interfacial DMI this is not a problem.
I also added a test to confirm the expected symmetry.

### CODE REVIEW

File: test/dmibulksymm1D.mx3 [hunk 1]
```diff
@@ -0,0 +1,43 @@
+/*
+	Symmetry test for bulk DMI in a 1D magnet with Neumann BC.
+	Rotating the entire system by 90° should give symmetric result.
+    This test is based on the proposed 1D standard problem of bulk BMI:
```
Comment (JonathanMaes) [inline test/dmibulksymm1D.mx3:4]: BMI typo?

Comment (ilateur) [reply test/dmibulksymm1D.mx3:4]: *DMI of course
